File: ios/RaceStat/RaceStatEmitter.m
//
//  TimerEventEmitter.m
//  F1Hub
//
//  Created by Anisimov Artem on 12.03.2024.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(RaceStatEmitter, RCTEventEmitter)

+ (bool)requiresMainQueueSetup {
  return NO;
}

RCT_EXTERN_METHOD(supportedEvents)

@end

File: ios/RaceStat/RaceStatEmitter.swift
//
//  TimerEventEmitter.swift
//  F1Hub
//
//  Created by Anisimov Artem on 12.03.2024.
//

import Foundation
import React

@objc(RaceStatEmitter)
class RaceStatEmitter: RCTEventEmitter {
  
  public static var emitter: RaceStatEmitter?
  
  override init() {
    super.init()
    RaceStatEmitter.emitter = self
  }
  
  override func supportedEvents() -> [String]! {
    return ["onStart", "onReset"]
  }
}
